parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -165,9 +178,65 @@ public void writeDouble(int repetitionLevel, double value) {
     }
   }
 
+  private static class FloatWriter extends UnboxedWriter<Float> {
+    private final ColumnDescriptor desc;
+    private long nanCount;
+
+    private FloatWriter(ColumnDescriptor desc, Type.ID id) {
+      super(desc, id);
+      this.desc = desc;
```

Comment:
The descriptor isn't used, so I don't think it is needed. This class should keep the `int id` only, I think.

Reply:
Not sure why I preserved it, thanks for pointing out!